### PR TITLE
PREAPPS-2941: Tags | Display tags in side bar

### PR DESCRIPTION
**Pull request by avadhutprabhudesai**

### CODE REVIEW

File: src/batch-client/index.ts [hunk 1]
```diff
@@ -707,6 +708,20 @@ export class ZimbraBatchClient {
 			namespace: Namespace.Account
 		});
 
+	public getTag = () =>
+		this.jsonRequest({
+			name: 'GetTag',
+			namespace: Namespace.Mail
+		}).then(({ tag }) => {
```
Comment (yashrajpchavda) [inline src/batch-client/index.ts]: can we write this like,

`}).then(({ tag }) => tag.map(normalize(Tag)) || []);`

File: src/schema/schema.ts [hunk 2]
```diff
@@ -177,7 +177,8 @@ export function createZimbraSchema(
 					client.shareInfo(variables as ShareInfoOptions),
 				taskFolders: client.taskFolders,
 				getWhiteBlackList: client.getWhiteBlackList,
-				getAppSpecificPasswords: client.getAppSpecificPasswords
+				getAppSpecificPasswords: client.getAppSpecificPasswords,
+				getTag: () => client.getTag()
```
Comment (itsmeprashant) [inline src/schema/schema.ts]: getTag: client.getTag

File: src/batch-client/index.ts (hunk 1 above)
Comment (avadhutprabhudesai) [reply src/batch-client/index.ts]: Is it good idea to modify function arguments directly?

Comment (yashrajpchavda) [reply src/batch-client/index.ts]: Umm, not sure if I understand you correctly, can you please elaborate?

Comment (avadhutprabhudesai) [reply src/batch-client/index.ts]: Got it. Fixed.

File: src/schema/schema.ts (hunk 2 above)
Comment (avadhutprabhudesai) [reply src/schema/schema.ts]: Fixed.

File: src/schema/schema.graphql [hunk 3]
```diff
@@ -1394,6 +1394,13 @@ type ContactListMember {
 	value: ID!
 }
 
+type Tag {
+	id: String
```
Comment (SantoshKShetty) [inline src/schema/schema.graphql]: Shouldn't this be `ID` ?

Comment (avadhutprabhudesai) [reply src/schema/schema.graphql]: Fixed.

**Pull request merged**